### PR TITLE
Stephen's December 2024 updates to macOS CI for 0.9.x

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation

Issues:
- Please list any related issues

Purpose:
- What is this pull request trying to do? Fix macOS builds for the 0.9.x branch
- What release is this for? 0.9.x
- Is there a project or milestone we should apply this to? 0.9.x

### CODE REVIEW

File: .github/workflows/macos-release.yml [hunk 1]
```diff
@@ -47,8 +47,13 @@ jobs:
       # - libpng
       # - libvorbis
       # - python
+      - name: Setup Python
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
```
Comment (BenjamenMeyer) [inline .github/workflows/macos-release.yml:52]: This might be what I need on master.

**Review by BenjamenMeyer — APPROVED** [review]

**Comment by stephengtuggy** [comment]
> We'll need to re-enable the Mac builds to see if this works

Oh, you guys have the Mac builds turned off for 0.9.x or something?

They worked on my fork of the repo. They should work here too.

File: .github/workflows/macos-release.yml (hunk 1 above)
Comment (stephengtuggy) [reply .github/workflows/macos-release.yml:52]: Probably. Along with some or all of the other changes I made in this PR.

File: engine/CMakeLists.txt [hunk 2]
```diff
@@ -33,6 +33,10 @@
 
 CMAKE_MINIMUM_REQUIRED(VERSION 3.16 FATAL_ERROR)
 
+IF (POLICY CMP0167)
+    CMAKE_POLICY (SET CMP0167 OLD)
```
Comment (stephengtuggy) [inline engine/CMakeLists.txt:37]: Even with the Setup Python action in place, the Mac builds still didn't work until I added this to CMakeLists.txt

File: engine/CMakeLists.txt [hunk 3]
```diff
@@ -115,6 +119,10 @@ UNSET(FFMPEG_FOUND)
 UNSET(OGRE_FOUND)
 UNSET(Boost_DIR)
 
+IF (CMAKE_SYSTEM_NAME STREQUAL Darwin)
+    SET(CMAKE_SHARED_LIBRARY_SUFFIX ".so")
```
Comment (stephengtuggy) [inline engine/CMakeLists.txt:123]: and added this, too

File: .github/workflows/macos-ci.yml [hunk 4]
```diff
@@ -55,18 +60,16 @@ jobs:
         if: ${{ matrix.homebrew-gl }}
         run: |
           brew install mesa mesa-glu freeglut
-          ln -s /usr/local/include/GL /usr/local/include/OpenGL
-          ln -s /usr/local/include/GL /usr/local/include/GLUT
-        # ln -s /usr/local/lib/libGL.dylib /usr/local/lib/libOpenGL.dylib
-        # find /usr/local/lib/ -iname '*gl*.dylib'
+          ln -s "$(brew --prefix)/include/GL" "$(brew --prefix)/include/OpenGL"
+          ln -s "$(brew --prefix)/include/GL" "$(brew --prefix)/include/GLUT"
```
Comment (stephengtuggy) [inline .github/workflows/macos-ci.yml:64]: Note that Homebrew installs stuff in a different default parent directory on Apple Silicon vs. x86_64. And the Homebrew directory can be changed from the default, also. Using `$(brew --prefix)` should take care of all of those cases.

File: .github/workflows/macos-ci.yml [hunk 5]
```diff
@@ -45,8 +45,13 @@ jobs:
       # - libpng
       # - libvorbis
       # - python
+      - name: Setup Python
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          python-version: 3.12.8
```
Comment (stephengtuggy) [inline .github/workflows/macos-ci.yml:51]: This specific Python version worked. 3.13 did not work.

File: .github/workflows/macos-ci.yml [hunk 6]
```diff
@@ -45,8 +45,13 @@ jobs:
...
+      
       - name: Install dependencies using homebrew
-        run: brew install boost-python3 gtk+3 gtkglext sdl
+        run: brew install boost-python3 gtk+3 gtkglext sdl libvorbis
```
Comment (stephengtuggy) [inline .github/workflows/macos-ci.yml:54]: Apparently, libvorbis is no longer preinstalled on the latest macOS runners. We have to install it ourselves now.

File: .github/workflows/macos-release.yml [hunk 7]
```diff
@@ -106,9 +109,12 @@ jobs:
           name: test_results_xml
           path: ${{github.workspace}}/build/test-results/**/*.xml
 
+      - name: Package it
+        run: cpack
+
```
Comment (stephengtuggy) [inline .github/workflows/macos-release.yml:114]: Please note! We are now running CPack, which (hopefully) will produce the installer package(s).

File: .github/workflows/macos-ci.yml [hunk 8]
```diff
@@ -21,8 +21,8 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          #- macos-11
-          - macos-12
+          - macos-13
+          # - macos-14
```
Comment (stephengtuggy) [inline .github/workflows/macos-ci.yml:25]: On macOS 14, the CI got as far as 50% through compiling the code, but then erred out at that point. I decided to leave macOS 14 for later.

File: .github/workflows/macos-ci.yml (hunk 5 above)
Comment (royfalk) [reply .github/workflows/macos-ci.yml:51]: I'm fine with this for now, but not supporting newer versions is a bad long term strategy. We should document this.

File: engine/CMakeLists.txt (hunk 2 above)
Comment (royfalk) [reply engine/CMakeLists.txt:37]: We're trailing behind CMake here. I'm not wild about that, as we're creating technical debt. However, I don't see another way, as we're resource constrained.

**Review by royfalk — APPROVED** [review]
First, let me welcome you back. You were missed. 

I'll document the two issues with python 3.13 and CMP0107. I'm more available the next few months, so there's a good chance I could get to it.

**Comment by royfalk** [comment]
Documented the issues in #933  and #934.

File: .github/workflows/macos-ci.yml (hunk 5 above)
Comment (stephengtuggy) [reply .github/workflows/macos-ci.yml:51]: @royfalk Agreed.

File: engine/CMakeLists.txt (hunk 2 above)
Comment (stephengtuggy) [reply engine/CMakeLists.txt:37]: @royfalk agreed. Also note that in order to get rid of this bit of technical debt, I think we'll have to remove support for CMake versions 3.29 and earlier.

**Comment by stephengtuggy** [comment]
> First, let me welcome you back. You were missed.
> 
> I'll document the two issues with python 3.13 and CMP0107. I'm more available the next few months, so there's a good chance I could get to it.

Thanks. It's good to be back. Still not sure how much I'll be able to be involved long-term, but I'm happy to help out a bit around the edges as I'm able.

**Pull request merged**